cmd/server/main.go

```diff
@@ -1,3 +1,170 @@
 package main
 
-func main() {}
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type gauge float64
+type counter int64
+
+type MemStorage struct {
+	Gauges   map[string]gauge
+	Counters map[string]counter
+}
+
+func (m *MemStorage) SetValue(t, name, value string) error {
+	switch t {
+	case "gauge":
+		i, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			return err
+		}
+		m.Gauges[name] = gauge(i)
+	case "counter":
+		i, err := strconv.Atoi(value)
+		if err != nil {
+			return err
+		}
+		v, ok := m.Counters[name]
+		if ok {
+			m.Counters[name] = v + counter(i)
+		} else {
+			m.Counters[name] = counter(i)
+		}
```

Comment:
Можно оптимизировать апдейт счетчика.

```suggestion
                m.Counters[name] += counter(i)
```